src/proxy.py

```diff
@@ -50,30 +55,58 @@ def __init__(self, loop):
 
         logging.info('HTTP REQUEST ' + str(request))
 
-        # Check if the request is on the blacklist or whitelist
-        if self.whitelist.check(request):
-            logging.info('Request is on the whitelist')
-        elif self.blacklist.check(request):
-            logging.info('Request is on the blacklist')
-            writer.write(b'HTTP/1.1 404 Not Found\n\n')
+        # this is where we should reject requests that come from unauthorized ip's
+        request_allowed_acl = False
+        try:
+            ip_address = writer.get_extra_info('peername')[0]
+
+            if ip_address is None:
+                logging.info('Could not get remote IP address')
+                writer.write(b'HTTP/1.1 403 Forbidden\n\n')
+                writer.close()
```

Comment:
We should return early from disallowed requests instead of storing state in the `request_allowed_acl` variable. Otherwise, the code becomes quite convoluted. So, I would return here.